x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/use_attack_discovery/index.tsx

```diff
@@ -151,37 +159,60 @@ export const useAttackDiscovery = ({
   }, [connectorId, pollData]);
 
   /** The callback when users click the Generate button */
-  const fetchAttackDiscoveries = useCallback(async () => {
-    try {
-      if (requestBody.apiConfig.connectorId === '' || requestBody.apiConfig.actionTypeId === '') {
-        throw new Error(CONNECTOR_ERROR);
-      }
-      setLoadingConnectorId?.(connectorId ?? null);
-      // sets isLoading to true
-      setPollStatus('running');
-      setIsLoadingPost(true);
-      setApproximateFutureTime(null);
-      // call the internal API to generate attack discoveries:
-      const rawResponse = await http.fetch('/internal/elastic_assistant/attack_discovery', {
-        body: JSON.stringify(requestBody),
-        method: 'POST',
-        version: ELASTIC_AI_ASSISTANT_INTERNAL_API_VERSION,
-      });
-      setIsLoadingPost(false);
-      const parsedResponse = AttackDiscoveryPostResponse.safeParse(rawResponse);
-
-      if (!parsedResponse.success) {
-        throw new Error('Failed to parse the response');
+  const fetchAttackDiscoveries = useCallback(
+    async (options: FetchAttackDiscoveriesOptions | undefined) => {
+      try {
+        if (options?.size != null) {
+          setAlertsContextCount(options.size);
+        }
+
+        const end = options?.end;
+        const filter = !isEmpty(options?.filter) ? options?.filter : undefined;
+        const start = options?.start;
+
+        const bodyWithOverrides = {
+          ...requestBody,
+          end,
+          filter,
+          size,
+          start,
+        };
+
+        if (
+          bodyWithOverrides.apiConfig.connectorId === '' ||
+          bodyWithOverrides.apiConfig.actionTypeId === ''
+        ) {
+          throw new Error(CONNECTOR_ERROR);
+        }
+        setLoadingConnectorId?.(connectorId ?? null);
+        // sets isLoading to true
+        setPollStatus('running');
+        setIsLoadingPost(true);
+        setApproximateFutureTime(null);
+
+        // call the internal API to generate attack discoveries:
+        const rawResponse = await http.post('/internal/elastic_assistant/attack_discovery', {
+          body: JSON.stringify(bodyWithOverrides),
+          method: 'POST',
```

Comment:
NIT: I believe we don't need `method: 'POST'` when we use `http.post` explicitly.

Reply:
Resolved by commit https://github.com/elastic/kibana/pull/205070/commits/4f83f49fe4476e388e8e3ce06f52327b68241249